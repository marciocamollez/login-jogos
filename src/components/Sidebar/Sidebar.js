import React, { useContext } from "react";
import styles from "./Sidebar.module.scss";

import Offcanvas from "react-bootstrap/Offcanvas";
import Button from "react-bootstrap/Button";
import InputGroup from "react-bootstrap/InputGroup";
import Form from "react-bootstrap/Form";

import { BsSearch } from "react-icons/bs";
import { BiHome } from "react-icons/bi";
import { FiUser } from "react-icons/fi";
import { AiOutlineRight } from "react-icons/ai";

import { Context } from "../../Context/AuthContext";
import { Link } from "react-router-dom";
import Header from "../Header/Header";

function Sidebar() {
  const { handleClose, show } = useContext(Context) || {};

  return (
    <div data-testid="test-sidemenu">
      <Header />

      <Offcanvas show={show} onHide={handleClose}>
        <Offcanvas.Header closeButton className={styles.border}>
          <Offcanvas.Title></Offcanvas.Title>
        </Offcanvas.Header>
        <Offcanvas.Body>
          <InputGroup className="mb-3">
            <Button id="button-addon1" className={styles.sidebarBtn}>
              <BsSearch />
            </Button>
            <Form.Control
              aria-label="Example text with button addon"
              aria-describedby="basic-addon1"
              placeholder="BUSCAR"
              className={styles.sidebarInput}
            />
          </InputGroup>

          <nav>
            <ul>
              <li>
                <Link to="/dashboard">
                  <span>
                    <BiHome />
                    Home
                  </span>
                  <span>
                    <AiOutlineRight />
                  </span>
                </Link>
              </li>
              <li>
                <Link to="/dashboard">
                  <span>
                    <FiUser />
                    Perfil
                  </span>
                  <span>
                    <AiOutlineRight />
                  </span>
                </Link>
              </li>
            </ul>
          </nav>
        </Offcanvas.Body>
      </Offcanvas>
    </div>
  );
}

export default Sidebar;
