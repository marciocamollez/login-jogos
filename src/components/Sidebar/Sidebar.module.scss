@import "../../styles/Variables.scss";

.border{
    border-bottom: 1px solid $lightGray;
}

.form-control{
    background-color: $lightGray;
}


.sidebarInput {
    background-color: $lightGray !important; 
    border: none !important;
}
.sidebarBtn{
    background-color: $lightGray !important;
    border: none !important;
    color: #9C9C9C !important;

    &:hover{
        background-color: $greenHover !important;
        color: #FFF !important;
    }
}



nav{
    ul{
        display: flex;
        flex-direction: column;
        padding: 0 20px;
        margin: 0;

        li{
            list-style: none;
            margin-bottom: 15px;
    
            a{
                display: flex;
                justify-content: space-between;
                text-transform: uppercase;
                color: $navyBlue;
                text-decoration: none;

                &:hover{
                    color: $greenHover;
                }

                svg{
                    margin-right: 10px;
                }
            }
        }
    }
}



