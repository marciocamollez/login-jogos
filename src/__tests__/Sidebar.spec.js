import { render, screen } from "@testing-library/react";
import Sidebar from "../components/Sidebar/Sidebar";
import { MemoryRouter } from "react-router-dom";

describe("Sidebar Component", () => {
  test("should render header div", () => {
    render(<Sidebar />, { wrapper: MemoryRouter });
    const el = screen.getByTestId("test-header");
    expect(el).toBeInTheDocument();
    expect(el).toMatchSnapshot();
    expect(el).toBeTruthy();
  });
});
