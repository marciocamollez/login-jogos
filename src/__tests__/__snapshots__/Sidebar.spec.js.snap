// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should render sidebar menu 1`] = `
<div
  data-testid="test-sidemenu"
>
  <header
    class="head"
    data-testid="test-header"
  >
    <div>
      <button
        class="btnMenuHeader"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </button>
      <a
        href="/dashboard"
      >
        <img
          alt="My Collection"
          src="/logo.png"
          title="My Collection"
        />
      </a>
    </div>
    <div>
      <button
        class="btnAddToCart"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="position-absolute top-10 start-90 translate-middle badge rounded-pill bg-danger"
        >
          4
          <span
            class="visually-hidden"
          >
            products
          </span>
        </span>
      </button>
      <a
        href="/dashboard"
      >
        <button
          class="btnProfile"
          type="button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
        </button>
      </a>
    </div>
  </header>
</div>
`;
